addons/signs/src/main/java/xyz/jpenilla/squaremap/addon/signs/data/CustomIcons.java

```diff
@@ -0,0 +1,94 @@
+package xyz.jpenilla.squaremap.addon.signs.data;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+
+import javax.imageio.ImageIO;
+
+import org.bukkit.plugin.java.JavaPlugin;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import xyz.jpenilla.squaremap.api.Key;
+import xyz.jpenilla.squaremap.api.SquaremapProvider;
+
+public final class CustomIcons {
+    private static final String KEY_PREFIX = "custom_";
+    private static final String CUSTOM_ICONS_DIR = "customicons";
+
+    private List<Key> keys = new ArrayList<>();
+
+    public static CustomIcons register(final @NonNull JavaPlugin plugin) {
+        CustomIcons out = new CustomIcons();
+
+        final File iconDir = new File(plugin.getDataFolder(), CUSTOM_ICONS_DIR);
+        iconDir.mkdir();
+
+        final File[] iconFiles = iconDir.listFiles();
+        if (iconFiles == null) {
+            plugin.getLogger().log(
+                    Level.WARNING,
+                    "squaremap-signs: unable to read customicons directory at \"{0}\"",
```

Comment:
The plugin logger is already prefixed; Also should use the slf4j logger i.e.
`plugin.getSLF4JLogger().warn("unable to read customicons directory at \"{}\"", iconDir.getAbsolutePath());`

Reply:
v2 switches to SLF4J, which is much nicer. But on my builds, the plugin logger does not appear to be prefixed:

```txt
[18:28:16] [Server thread/INFO]: Enabling squaremap-signs v1.0.0-SNAPSHOT
[18:28:17] [Server thread/WARN]: unable to read custom icon "plugins/squaremap-signs/customicons/not-an-icon" as an image
```

Reply:
Do you have a custom log4j config? This is what I get with default Paper when I add `this.getSLF4JLogger().info("test");` in `onEnable`:
```
[12:24:41 INFO]: [squaremap-signs] Enabling squaremap-signs v1.0.0-SNAPSHOT
[12:24:41 INFO]: [squaremap-signs] test
```

Reply:
Oh, I do! I'm using the [docker.io/itzg/minecraft-server](https://github.com/itzg/docker-minecraft-server) image to host Minecraft. That injects a custom `log4j2.xml` file—probably to patch the dreaded CVE-2021-44228.

That would explain it.